dangerfile.ts

```diff
@@ -90,25 +89,40 @@ function chunked<T>(arr: T[], size: number): T[][] {
 }
 
 const unformatted = [];
+const dprintErrors = [];
 const allFiles = [...danger.git.created_files, ...danger.git.modified_files];
 // We batch this in chunks to avoid hitting max arg length issues.
 for (const files of chunked(allFiles, 50)) {
     const result = cp.spawnSync(
         process.execPath,
-        ["node_modules/dprint/bin.js", "check", ...files],
+        ["node_modules/dprint/bin.js", "check", "--list-different", ...files],
         { encoding: "utf8", maxBuffer: 100 * 1024 * 1024 },
     );
-    if (result.status !== 0) {
+    // https://dprint.dev/cli/#exit-codes
+    if (result.status === 20) {
         for (const line of result.stdout.split(/\r?\n/)) {
-            const match = stripAnsi(line).match(/^from (.*):$/);
-            if (match) {
-                unformatted.push(path.relative(process.cwd(), match[1]));
+            if (line) {
+                unformatted.push(path.relative(process.cwd(), line));
             }
         }
+    } else if (status !== 0) {
+        dprintErrors.push(result.stderr.trim());
     }
 }
 
-if (unformatted.length > 0) {
+if (dprintErrors.length > 0) {
+    fail("dprint failed to execute");
+
+    const message = [
+        "## Formatting errors",
+        "",
+        "```",
+        ...dprintErrors.join("\n\n"),
```

Comment:
It might be worthwhile to escape this somehow to ensure the errors don't escape out of the codeblock somehow and someone gets the bot to display something. That said, probably wouldn't ever happen.

Reply:
Yeah, I'm just not sure how I'd escape an ending triple backtick inside of a code block if one were to appear in the right place. At best I could use a silly number of backticks (since the number of them doesn't matter as long as they match).

Reply:
Used like 10 backticks, which "ought to be enough for anyone"...

Reply:
That escaping only works for inline backticks, I'm afraid.